timer/timer.c

```diff
@@ -37,13 +38,19 @@ static ssize_t inter_show(struct class *class, struct class_attribute *attr,
 static ssize_t time_show(struct class *class, struct class_attribute *attr,
 			 char *buffer)
 {
-	static ktime_t time = 0;
-	static s32 hr_sec = 0;
-	static s64 hr_nsec = 0;
+	ktime_t time = 0;
+	char hr_sec[MAX_HR_BUFF] = "";
+	size_t len = 0;
 	time = hrtimer_cb_get_time(&hr_timer);
-	hr_sec = time / NSEC_PER_SEC;
-	hr_nsec = time % NSEC_PER_SEC;
-	sprintf(buffer, "inter = %u s %lld ns\n", hr_sec, hr_nsec);
+	len = snprintf(hr_sec, MAX_HR_BUFF, "%lld", time);
+	if (len > DOT_POS && len < MAX_HR_BUFF - 1) {
+		size_t i;
+		for (i = len - 1; i > len - DOT_POS; --i) {
+			hr_sec[i + 1] = hr_sec[i];
+		}
+		hr_sec[len - DOT_POS] = '.';
+	}
+	sprintf(buffer, "time = %s s\n", hr_sec);
```

Comment:
What is the reason for timer printing redesign?

Reply:
ktime_t store value in nanoseconds. So, for getting time string in seconds, I have used this string conversion. Before I was getting the same via division and modulo operations

Reply:
I have two questions about the implementation of boot option (via DTS). What kind of driver data structure (like "platform_driver" or "i2c_driver") should I use? And which .dtsi and .dts files are proper for changing?

Reply:
Can you simplify this code? I think previous implementation was better.